nemo-physical/src/dictionary/datavalue_dictionary.rs

```diff
@@ -115,4 +165,52 @@ pub trait DvDict: Debug + ByteSized {
 
     /// Returns true if the dictionary contains any marked elements (see [DvDict::mark_dv]).
     fn has_marked(&self) -> bool;
+
+    /// Returns a mutable object that implements this trait as a [std::any::Any]. Dynamic
+    /// trait objects cannot be used as [std::any::Any], but concrete implementations
+    /// can provide a (dynamic) method that accomplishes this. This is a pattern that
+    /// can then be used for casting the dynamic object into a concrete type.
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
```

Comment:
These methods should probably not be part of this trait, since they can trivially be implemented for (almost) any type.

Instead the following is more scalable (no need to implement as_any for every dictionary type):

```
trait AsAny {
  fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
  fn as_any(&self) -> &dyn std::any::Any;
}

impl<T: 'static> AsAny for T {
  fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
    self
  }

  fn as_any(&self) -> &dyn std::any::Any {
    self
  }
}
```

With that the `as_any` method is available for all types where applicable.

Reply:
Nice solution. Will try.